core/primitives/src/profile_data_v2.rs

```diff
@@ -48,32 +48,37 @@ impl ProfileDataV2 {
     /// This is used to display old gas profiles on the RPC API and in debug output.
     pub fn legacy_action_costs(&self) -> Vec<(&'static str, Gas)> {
         vec![
-            ("CREATE_ACCOUNT", self.data[0]),
-            ("DELETE_ACCOUNT", self.data[1]),
-            ("DEPLOY_CONTRACT", self.data[2]), // contains per byte and base cost
-            ("FUNCTION_CALL", self.data[3]),   // contains per byte and base cost
-            ("TRANSFER", self.data[4]),
-            ("STAKE", self.data[5]),
-            ("ADD_KEY", self.data[6]), // contains base + per byte cost for function call keys and full access keys
-            ("DELETE_KEY", self.data[7]),
-            ("NEW_DATA_RECEIPT_BYTE", self.data[8]), // contains the per-byte cost for sending back a data dependency
-            ("NEW_RECEIPT", self.data[9]), // contains base cost for data receipts and action receipts
+            ("CREATE_ACCOUNT", Gas::from_gas(self.data[0])),
+            ("DELETE_ACCOUNT", Gas::from_gas(self.data[1])),
+            ("DEPLOY_CONTRACT", Gas::from_gas(self.data[2])), // contains per byte and base cost
+            ("FUNCTION_CALL", Gas::from_gas(self.data[3])),   // contains per byte and base cost
+            ("TRANSFER", Gas::from_gas(self.data[4])),
+            ("STAKE", Gas::from_gas(self.data[5])),
+            ("ADD_KEY", Gas::from_gas(self.data[6])), // contains base + per byte cost for function call keys and full access keys
+            ("DELETE_KEY", Gas::from_gas(self.data[7])),
+            ("NEW_DATA_RECEIPT_BYTE", Gas::from_gas(self.data[8])), // contains the per-byte cost for sending back a data dependency
+            ("NEW_RECEIPT", Gas::from_gas(self.data[9])), // contains base cost for data receipts and action receipts
         ]
     }
 
     pub fn action_gas(&self) -> u64 {
-        self.legacy_action_costs().iter().map(|(_name, cost)| *cost).fold(0, u64::saturating_add)
+        self.legacy_action_costs()
+            .iter()
+            .map(|(_name, cost)| *cost)
+            .fold(Gas::from_gas(0), |acc: Gas, cost: Gas| acc.saturating_add(cost))
+            .as_gas()
     }
 
     /// Test instance with unique numbers in each field.
     pub fn test() -> Self {
         let mut profile_data = ProfileDataV2::default();
         let num_legacy_actions = 10;
         for i in 0..num_legacy_actions {
-            profile_data.data.0[i] = i as Gas + 1000;
+            profile_data.data.0[i] =
+                Gas::from_gas(i as u64).checked_add(Gas::from_gas(1000)).unwrap().as_gas();
```

Comment:
I think it does not make a ton of sense to do all these Gas manipulations here, so I'd just revert to:

```suggestion
            profile_data.data.0[i] = i as u64 + 1000;
```